From 42a4d47df0eeeef41b18d7aae04bdc3d3a80eef3 Mon Sep 17 00:00:00 2001
From: Marcelo Leitner <mleitner@redhat.com>
Date: Tue, 21 Oct 2014 16:21:07 -0400
Subject: [net] netfilter: fix several Kconfig problems in NF_LOG_*

Message-id: <20141021162107.22888.99086.stgit@localhost.localdomain>
Patchwork-id: 98013
O-Subject: [RHEL7 net PATCH 116/124] netfilter: fix several Kconfig problems in NF_LOG_*
Bugzilla: 1148041 1155088
RH-Acked-by: Florian Westphal <fwestpha@redhat.com>

Backport of upstream commit: c1878869c0c8e0def3df5397155f369442ce4e06
Backport notes:
We don't have NF_LOG_ARP..

 commit c1878869c0c8e0def3df5397155f369442ce4e06
 Author: Pablo Neira Ayuso <pablo@netfilter.org>
 Date:   Sat Jun 28 18:39:01 2014 +0200

    netfilter: fix several Kconfig problems in NF_LOG_*

    warning: (NETFILTER_XT_TARGET_LOG) selects NF_LOG_IPV6 which has unmet direct dependencies (NET && INET && IPV6 && NETFILTER && IP6_NF_IPTABLES && NETFILTER_ADVANCED)
    warning: (NF_LOG_IPV4 && NF_LOG_IPV6) selects NF_LOG_COMMON which has unmet direct dependencies (NET && INET && NETFILTER && NF_CONNTRACK)

    Fixes: 83e96d4 ("netfilter: log: split family specific code to nf_log_{ip,ip6,common}.c files")
    Reported-by: Fengguang Wu <fengguang.wu@intel.com>
    Signed-off-by: Pablo Neira Ayuso <pablo@netfilter.org>

Signed-off-by: Marcelo Ricardo Leitner <mleitner@redhat.com>
Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/ipv4/netfilter/Kconfig b/net/ipv4/netfilter/Kconfig
index 40d0ae4..aed0f45 100644
--- a/net/ipv4/netfilter/Kconfig
+++ b/net/ipv4/netfilter/Kconfig
@@ -36,6 +36,11 @@ config NF_CONNTRACK_PROC_COMPAT
 
    If unsure, say Y.
 
+config NF_LOG_IPV4
+ tristate "IPv4 packet logging"
+ default m if NETFILTER_ADVANCED=n
+ select NF_LOG_COMMON
+
 config NF_TABLES_IPV4
  depends on NF_TABLES
  tristate "IPv4 nf_tables support"
@@ -178,11 +183,6 @@ config IP_NF_TARGET_ULOG
 
    To compile it as a module, choose M here.  If unsure, say N.
 
-config NF_LOG_IPV4
- tristate "IPv4 packet logging"
- default m if NETFILTER_ADVANCED=n
- select NF_LOG_COMMON
-
 # NAT + specific targets: nf_conntrack
 config NF_NAT_IPV4
  tristate "IPv4 NAT"
diff --git a/net/ipv6/netfilter/Kconfig b/net/ipv6/netfilter/Kconfig
index f36ea3f..24c535f 100644
--- a/net/ipv6/netfilter/Kconfig
+++ b/net/ipv6/netfilter/Kconfig
@@ -55,6 +55,11 @@ config NFT_REJECT_IPV6
  default NFT_REJECT
  tristate
 
+config NF_LOG_IPV6
+ tristate "IPv6 packet logging"
+ depends on NETFILTER_ADVANCED
+ select NF_LOG_COMMON
+
 config IP6_NF_IPTABLES
  tristate "IP6 tables support (required for filtering)"
  depends on INET && IPV6
@@ -227,11 +232,6 @@ config IP6_NF_SECURITY
 
          If unsure, say N.
 
-config NF_LOG_IPV6
- tristate "IPv6 packet logging"
- depends on NETFILTER_ADVANCED
- select NF_LOG_COMMON
-
 config NF_NAT_IPV6
  tristate "IPv6 NAT"
  depends on NF_CONNTRACK_IPV6
diff --git a/net/netfilter/Kconfig b/net/netfilter/Kconfig
index d7c6834..f6cce87 100644
--- a/net/netfilter/Kconfig
+++ b/net/netfilter/Kconfig
@@ -46,6 +46,9 @@ config NF_CONNTRACK
 
    To compile it as a module, choose M here.  If unsure, say N.
 
+config NF_LOG_COMMON
+ tristate
+
 if NF_CONNTRACK
 
 config NF_CONNTRACK_MARK
@@ -359,9 +362,6 @@ config NETFILTER_NETLINK_QUEUE_CT
    If this option is enabled, NFQUEUE can include Connection Tracking
    information together with the packet is the enqueued via NFNETLINK.
 
-config NF_LOG_COMMON
- tristate
-
 config NF_NAT
  tristate
 
@@ -756,9 +756,7 @@ config NETFILTER_XT_TARGET_LED
 
 config NETFILTER_XT_TARGET_LOG
  tristate "LOG target support"
- select NF_LOG
- select NF_LOG_IPV4
- select NF_LOG_IPV6
+ depends on NF_LOG_IPV4 && NF_LOG_IPV6
  default m if NETFILTER_ADVANCED=n
  help
    This option adds a `LOG' target, which allows you to create rules in
-- 
1.7.1